src/Bicep.Core/CoreResources.resx

```diff
@@ -520,4 +520,14 @@
   <data name="UseSafeAccessRule_MessageFormat" xml:space="preserve">
     <value>The syntax can be simplified by using the safe access (.?) operator.</value>
   </data>
-</root>
\ No newline at end of file
+  <data name="WhatIfShortCircuitingRuleDescription" xml:space="preserve">
+    <value>Runtime values cannot be used to determine resource IDs</value>

```

Comment:
```suggestion
    <value>Runtime values should not be used to determine resource IDs</value>
```

[nit]: Since the ARM engine does allow this, it's more a question of "should" than "could."